### PR TITLE
asserts: sort by revision with Sort interface

### PR DESCRIPTION
**Pull request by stolowski**
Followup to #9732, implement and use byRevision for revision sorting in the error handlers (thanks @bboozzoo for suggestion). Also a tiny cleanup of -= from golint.

### CODE REVIEW

**Review by bboozzoo — COMMENTED** [review]
The last patch LGTM

**Review by bboozzoo — APPROVED** [review]
LGTM

File: asserts/snapasserts/validation_sets.go [hunk 1]
```diff
@@ -304,7 +309,7 @@ func (v *ValidationSets) addSnap(sn *asserts.ValidationSetSnap, validationSetKey
 	// this counts really different revisions or invalid
 	ndiff := len(cs.revisions)
 	if _, ok := cs.revisions[unspecifiedRevision]; ok {
-		ndiff -= 1
```
Comment (zyga) [inline asserts/snapasserts/validation_sets.go:307]: Unrelated but meh :)

**Review by zyga — APPROVED** [review]
Looks good, I'd add a unit test just to be on the safe side of the coverage delta.

**Review by mvo5 — APPROVED** [review]
Thank you, looks fine

**Pull request merged**